bin/cli.js

```diff
@@ -104,15 +143,19 @@ async function main() {
         fs.mkdirSync(dir);
       }
     }
+  }
 
-    fs.writeFileSync(outputFile, result, "utf8");
+  // if there are multiple specs, ensure that output is a directory
+  if (inputSpecPaths.length > 1 && output === OUTPUT_FILE && !fs.lstatSync(cli.flags.output).isDirectory()) {
+    errorAndExit(
+      `❌ When specifying a glob matching multiple input specs, you must specify a directory for generated type definitions.`
+    );
+  }
 
-    const timeEnd = process.hrtime(timeStart);
-    const time = timeEnd[0] + Math.round(timeEnd[1] / 1e6);
-    console.log(green(`🚀 ${pathToSpec} -> ${bold(cli.flags.output)} [${time}ms]`));
-  } else {
-    // output option 2: stdout
-    process.stdout.write(result);
+  let result = "";
+  for (const specPath of inputSpecPaths) {
+    // append result returned for each spec
+    result += await generateSchema(specPath);
```

Comment:
This is clever. I hadn’t thought about the `stdout` case but this seems pretty handy.

Reply:
Oh wait—we’re logging to `stdout` elsewhere. Truth be told, I don’t know why we’re returning the `result` here; it doesn’t seem to be helpful (and possibly wasting memory?). Either way, this is fine to leave in this PR.

Reply:
Hey, I was initially confused as well about what the reason to return the result was. In user usage, the returned result doesn't matter, but I think it gets used in certain tests so it should be kept in.

Reply:
OK. Yeah leave it in for now, and I can revisit that post-merge.

Reply:
Alright! 
Rebasing and updating expected schemas fixed the broken test, so I think this PR should be ready now